azure/services/subnets/client.go

```diff
@@ -54,43 +61,113 @@ func newSubnetsClient(subscriptionID string, baseURI string, authorizer autorest
 }
 
 // Get gets the specified subnet by virtual network and resource group.
-func (ac *AzureClient) Get(ctx context.Context, resourceGroupName, vnetName, snName string) (network.Subnet, error) {
+func (ac *AzureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (result interface{}, err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "subnets.AzureClient.Get")
 	defer done()
 
-	return ac.subnets.Get(ctx, resourceGroupName, vnetName, snName, "")
+	return ac.subnets.Get(ctx, spec.ResourceGroupName(), spec.OwnerResourceName(), spec.ResourceName(), "")
```

Comment:
see above for comment on vnet resource group.

Reply:
this should be resolved now that spec.ResourceGroup is being set to the vnet RG?